pkg/datapath/linux/config/config.go

```diff
@@ -643,6 +643,9 @@ func (h *HeaderfileWriter) WriteNodeConfig(w io.Writer, cfg *datapath.LocalNodeC
 		cDefinesMap["SNAT_COLLISION_RETRIES"] = fmt.Sprintf("%d", nat.SnatCollisionRetries)
 
 		if option.Config.EnableBPFMasquerade {
+			if option.Config.EnableRemoteNodeSnat {
```

Comment:
should we log a warning if someone enables remote node snat without enabling bpf masquerade?

Reply:
Good point. I'd even just fatal in that case as that's not a config that makes sense. Without a signal, users might be confused as to why the flag is without effect on iptables-based masquerading.

(FYI, such an error should probably go in `daemon_main.go`; we have multiple such error cases.)

Reply:
Will add a warning in that case